test/test_optim.py

```diff
@@ -1844,10 +1844,11 @@ def closure():
                         "cuda" if capturable or fused else "cpu",
                     )
 
-            for _ in range(5):
+            for step in range(5):
+                factor = step + 1
                 optimizer.step(closure)
                 optimizer_cuda.step(closure)
-                self.assertEqual(params, params_cuda)
+                self.assertEqual(params, params_cuda, atol=factor * 2e-4, rtol=factor * 1e-4)
```

Comment:
where does 2e-4 and 1e-4 come from? They seem a bit large...
By the end of 5 steps we'd have atol of 1e-3.

Instead of just multiplying the tols, I'd prefer a solution using TensorTracker like in https://github.com/pytorch/pytorch/blob/main/test/test_optim.py#L735C9-L778